### PR TITLE
Only consider error-free scheduled posts for indicator in channel and RHS

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Previously I was checking if there is any scheduled post for the channel and showing the post box indicator accordingly. Now I show it only if there is an error-free scheduled post. I also considered showing only for future scheduled posts but since the backend process scheduled posts up tp 24 hours old, it is possible for a scheduled posts few hours in the past to still be processed.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-60944

#### Release Note
```release-note
None
```

### CODE REVIEW

File: webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts [hunk 1]
```diff
@@ -56,13 +56,18 @@ export function hasScheduledPostError(state: GlobalState, teamId: string) {
 }
 
 export function showChannelOrThreadScheduledPostIndicator(state: GlobalState, channelOrThreadId: string): ChannelScheduledPostIndicatorData {
-    const channelScheduledPosts = state.entities.scheduledPosts.byChannelOrThreadId[channelOrThreadId] || emptyList;
+    const allChannelScheduledPosts = state.entities.scheduledPosts.byChannelOrThreadId[channelOrThreadId];
+    const eligibleScheduledPosts = allChannelScheduledPosts.filter((scheduledPostId: string) => {
```
Comment (pvev) [inline webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts:60]:
```suggestion
    const eligibleScheduledPosts = allChannelScheduledPosts?.filter((scheduledPostId: string) => {
```

File: webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts [hunk 2]
```diff
@@ -56,13 +56,18 @@ export function hasScheduledPostError(state: GlobalState, teamId: string) {
...
+        const scheduledPost = state.entities.scheduledPosts.byId[scheduledPostId];
+        return !scheduledPost.error_code;
```
Comment (pvev) [inline webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts:62]: is there a type for the scheduledPost object? wondering why the snake case for error_code?

**Review by pvev — APPROVED** [review]
LGTM! just one question. Thanks @harshilsharma63 🎉

File: webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts (hunk 1 above)
Comment (harshilsharma63) [reply webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts:60]: AH so yeah that was the bug. I restored the original `|| emptyLIst` part so we don't need the ?. Just tha that empty list part was there before, thats why, essentially both are similar

File: webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts (hunk 2 above)
Comment (harshilsharma63) [reply webapp/channels/src/packages/mattermost-redux/src/selectors/entities/scheduled_posts.ts:62]: So all the types that for API responses have camel case, like this one

**Review by yasserfaraazkhan — APPROVED** [review]
the channel wont consider failed scheduled drafts when showing `See all` option

**Pull request merged**